### ISSUE TITLE
JupyterQ incompatible with Python 3.8?

### ISSUE THREAD

**Issue by kamchan0** [KxSystems/jupyterq#57]
I just installed the latest miniconda3 for windows 64bit. When I hit the command

`conda install -c kx jupyterq`
Then it says

```
UnsatisfiableError: The following specifications were found
to be incompatible with the existing python installation in your environment:

Specifications:

  - jupyterq -> python[version='>=3.5,<3.6.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0']

Your python: python=3.8

If python is on the left-most side of the chain, that's the version you've asked for.
When python appears to the right, that indicates that the thing on the left is somehow
not available for the python version you are constrained to. Note that conda will not
change your python version to a different minor version unless you explicitly specify
that.

The following specifications were found to be incompatible with your CUDA driver:

  - feature:/win-64::__cuda==9.1=0
  - feature:|@/win-64::__cuda==9.1=0

Your installed CUDA driver is: 9.1

```
Sounds like jupyterQ is incompatible with python 3.8? Can you update your package in conda?

**Comment by kamchan0** [comment]
Thanks for this but it is still actually the same. Do I miss some steps?

```
c:\xxx> conda install -c kx jupyterq
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: -
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed

UnsatisfiableError: The following specifications were found
to be incompatible with the existing python installation in your environment:

Specifications:

  - jupyterq -> python[version='>=3.5,<3.6.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0']

Your python: python=3.8

If python is on the left-most side of the chain, that's the version you've asked for.
When python appears to the right, that indicates that the thing on the left is somehow
not available for the python version you are constrained to. Note that conda will not
change your python version to a different minor version unless you explicitly specify
that.

The following specifications were found to be incompatible with your CUDA driver:

  - feature:/win-64::__cuda==9.1=0

Your installed CUDA driver is: 9.1
```

**Comment by cmccarthy1** [comment]
Hi @kamchan0 

We're presently having some issues deploying the windows builds to Conda, while releasing the jupyterq version we're also releasing an update to embedPy. We've been able to deploy Linux and Mac 3.8 versions so far for embedPy and are working on a fix for the windows deployments. I'll have a member of the team update here by EOD tomorrow with progress if we haven't been able to get a fix in place before then.

All the best,

Conor

**Comment by cmccarthy1** [comment]
Hi @kamchan0 

This is proving more difficult to change than had been expected. Python 3.8 modified how it provides support for using python in embedded settings in 3.8 as outlined [here](https://docs.python.org/3/whatsnew/3.8.html#debug-build-uses-the-same-abi-as-release-build), we're currently working on a fix but at present do not have a hard timeline on which we expect to have this updated. I'll revert here once we make progress on this matter.

All the best,

Conor

**Comment by cmccarthy1** [comment]
Issue closed with release 1.1.8, should now be available via conda install for mac, Linux and windows
